PixivConfig.py

```diff
@@ -206,12 +210,23 @@ class PixivConfig():
         ConfigItem("DownloadControl", "postProcessingCmd", ""),
     ]
 
-    proxy = {"http": "", "https": "", }
-
     def __init__(self):
         for item in self.__items:
             setattr(self, item.option, item.process_value(item.default))
-        self.proxy = {'http': self.proxyAddress, 'https': self.proxyAddress}
+
+    @property
+    def proxy(self):
+        value = getattr(self, "proxyAddress", None)
+        if not value:
+            return None
+        match = re.match(r"^(?:(http|socks[45])://)?(\d+\.\d+\.\d+\.\d+):(\d+)$", value)
```

Comment:
maybe can use this pattern `^(https|http|socks[45])://[\w.]+(:?\d*)$`
![image](https://user-images.githubusercontent.com/836867/142389850-f07b9ae1-8a66-4029-b0c2-1f2a081b1e7f.png)

Reply:
updated to allow `-` = `^(https|http|socks[45])://[\w.-]+(:?\d*)$`
![image](https://user-images.githubusercontent.com/836867/142390202-c375ecb3-b574-46c8-aeb4-5fc35ebe5d13.png)